compiler/plc_diagnostics/src/diagnostician.rs

```diff
@@ -147,27 +151,62 @@ impl DiagnosticAssessor for DefaultDiagnosticAssessor {
         match d {
             // improvements become warnings
             Diagnostic::ImprovementSuggestion { .. } => Severity::Warning,
+            _ if *d.get_type() == ErrNo::reference__unresolved => Severity::Critical,
             // everything else becomes an error
             _ => Severity::Error,
         }
     }
 }
 
 /// a diagnostics severity
-#[derive(Clone, Debug, PartialEq, Eq)]
+#[derive(Clone, Copy, Debug, PartialEq, Eq)]
 pub enum Severity {
+    Critical,
     Error,
     Warning,
-    _Info,
+    Info,
 }
 
 impl std::fmt::Display for Severity {
     fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
         let severity = match self {
+            Severity::Critical => "critical",
             Severity::Error => "error",
             Severity::Warning => "warning",
-            Severity::_Info => "info",
+            Severity::Info => "info",
         };
         write!(f, "{severity}")
     }
 }
+
+impl PartialOrd for Severity {
+    fn partial_cmp(&self, other: &Self) -> Option<std::cmp::Ordering> {
+        Some(self.cmp(other))
+    }
+}
+
+impl Ord for Severity {
+    fn cmp(&self, other: &Self) -> std::cmp::Ordering {
```

Comment:
I find this a pretty complicated way to compare
couldnt you just assign a weight to the enum? and probably you can then derive Ord and ParitalOrd

Reply:
probably, somehow didn't think of that